File: app/src/main/java/com/waveneuro/ui/introduction/splash/SplashActivity.java
package com.waveneuro.ui.introduction.splash;

import android.graphics.Color;
import android.os.Bundle;
import android.view.View;

import androidx.core.splashscreen.SplashScreen;
import androidx.lifecycle.Observer;

import com.waveneuro.R;
import com.waveneuro.ui.base.BaseActivity;
import com.waveneuro.ui.dashboard.DashboardCommand;
import com.waveneuro.ui.user.login.LoginCommand;

import javax.inject.Inject;

public class SplashActivity extends BaseActivity {


    @Inject
    LoginCommand loginCommand;
    @Inject
    DashboardCommand dashboardCommand;

    @Inject
    SplashViewModel splashViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        SplashScreen splashScreen = SplashScreen.installSplashScreen(this);
        super.onCreate(savedInstanceState);
        getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
        getWindow().setStatusBarColor(Color.TRANSPARENT);
        activityComponent().inject(this);
        setContentView(R.layout.activity_splash);
        setView();
        setObserver();

        this.splashViewModel.processEvent(new SplashViewEvent.Start());
    }

    private void setView() {

    }

    private void setObserver() {
        this.splashViewModel.getData().observe(this, splashViewStateObserver);
        this.splashViewModel.getViewEffect().observe(this, splashViewEffectObserver);
    }

    Observer<SplashViewState> splashViewStateObserver = viewState -> {

    };

    Observer<SplashViewEffect> splashViewEffectObserver = viewEffect -> {
        if (viewEffect instanceof SplashViewEffect.Login) {
            launchLoginScreen();
        } else if (viewEffect instanceof SplashViewEffect.Home) {
            launchHomeScreen();
        }
    };


    private void launchLoginScreen() {
        loginCommand.navigate();
    }

    private void launchHomeScreen() {
        dashboardCommand.navigate();
    }
}

File: app/src/main/java/com/waveneuro/ui/introduction/splash/SplashActivity.kt
package com.waveneuro.ui.introduction.splash

import android.graphics.Color
import android.os.Bundle
import android.view.View
import androidx.activity.viewModels
import androidx.lifecycle.Observer
import com.waveneuro.databinding.ActivitySplashBinding
import com.waveneuro.ui.base.activity.BaseViewModelActivity
import com.waveneuro.ui.dashboard.DashboardActivity
import com.waveneuro.ui.introduction.splash.viewmodel.SplashViewModel
import com.waveneuro.ui.introduction.splash.viewmodel.SplashViewModelImpl
import com.waveneuro.ui.user.login.LoginActivity
import com.waveneuro.utils.ext.getAppComponent

class SplashActivity : BaseViewModelActivity<ActivitySplashBinding, SplashViewModel>() {

    override val viewModel: SplashViewModelImpl by viewModels {
        getAppComponent().splashViewModelFactory()
    }

    override fun initBinding(): ActivitySplashBinding =
        ActivitySplashBinding.inflate(layoutInflater)

    override fun onCreate(savedInstanceState: Bundle?) {
        window.decorView.systemUiVisibility = (View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                or View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN)
        window.statusBarColor = Color.TRANSPARENT
        super.onCreate(savedInstanceState)

        setObserver()
        viewModel.processEvent(SplashViewEvent.Start)
    }

    private fun setObserver() {
        viewModel.viewEffect.observe(this, Observer { viewEffect ->
            when (viewEffect) {
                is SplashViewEffect.Login -> launchLoginScreen()
                is SplashViewEffect.Home -> launchHomeScreen()
            }
        })
    }
    private fun launchLoginScreen() {
        startActivity(LoginActivity.newIntent(this))
        finish()
    }

    private fun launchHomeScreen() {
        startActivity(DashboardActivity.newIntent(this))
        finish()
    }

}